atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SerializableTransaction.java

```diff
@@ -795,31 +796,48 @@ private Transaction getReadOnlyTransaction(final long commitTs) {
                         afterStart.add(startTs);
                     }
                 }
-                Map<Long, Long> ret = Maps.newHashMap();
-                if (!afterStart.isEmpty()) {
-                    // We do not block when waiting for results that were written after our
-                    // start timestamp.  If we block here it may lead to deadlock if two transactions
-                    // (or a cycle of any length) have all written their data and all doing checks before committing.
-                    Map<Long, Long> afterResults = super.getCommitTimestamps(tableRef, afterStart, false);
-                    if (!afterResults.keySet().containsAll(afterStart)) {
-                        // If we do not get back all these results we may be in the deadlock case so we should just
-                        // fail out early.  It may be the case that abort more transactions than needed to break the
-                        // deadlock cycle, but this should be pretty rare.
-                        transactionOutcomeMetrics.markReadWriteConflict(tableRef);
-                        throw new TransactionSerializableConflictException("An uncommitted conflicting read was "
-                                + "written after our start timestamp for table " + tableRef + ".  "
-                                + "This case can cause deadlock and is very likely to be a read write conflict.");
-                    } else {
-                        ret.putAll(afterResults);
-                    }
-                }
+
+                boolean containsMyStartFinal = containsMyStart;
```

Comment:
???, extract the above to a method, with 2 output params (eurgh), the method returns `containsMyStart`